git/e2e/gitlab_test.go

```diff
@@ -0,0 +1,180 @@
+//go:build e2e
+// +build e2e
+
+/*
+// Copyright 2022 The Flux authors
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	"github.com/fluxcd/pkg/git"
+	"github.com/fluxcd/pkg/git/gogit"
+)
+
+const (
+	gitlabUsername = "root"
+	gitlabHTTPHost = "http://127.0.0.1:8080"
+	gitlabSSHHost  = "ssh://git@127.0.0.1:2222"
+)
+
+func TestGitLabE2E(t *testing.T) {
+	privateToken := os.Getenv("GITLAB_PAT")
```

Comment:
I would move the variable names to constants, makes it easier to spot what requires configuration.